webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/Boot.scala

```diff
@@ -357,11 +358,58 @@ class Boot extends Loggable {
 
     // By default Lift redirects to login page when a comet request's session changes
     // which happens when there is a connection to the same server in another tab.
-    // Do nothing instead, as at allows to keep open tabs context until we get the new cookie
+    // Do nothing instead, as it allows to keep open tabs context until we get the new cookie
     // This does not affect security as it is only a redirection anyway and did not change
     // the session itself.
     LiftRules.noCometSessionCmd.default.set(() => JsCmd.unitToJsCmd(()))
 
+    // Store access time for user idle tracking in each non-comet request
+    // Required as standard idle timeout includes comet requests,
+    // which practically means that an open page never expires.
+    LiftRules.onBeginServicing.append((r: Req) => {
+      ApplicationLogger.warn(s"URL: ${r.uri}")
+      // This filters out lift-related XHR only, which is exactly what we want
+      if (r.standardRequest_?) {
+        ApplicationLogger.warn(s"URL: ${r.uri}: lastNonCometAccessedTime")
+        LiftRules.getLiftSession(r).httpSession
+          .foreach(s => s.setAttribute("lastNonCometAccessedTime", millis))
+      }
+    })
+
+    // Custom session expiration that ignores comet requests
+    val IdleSessionTimeout = (sessions: Map[String, SessionInfo], delete: SessionInfo => Unit) => {
+      for ((id, info) <- sessions) {
+        info.session.httpSession.foreach(s => {
+          s.attribute("lastNonCometAccessedTime") match {
+            case lastNonCometAccessedTime: Long =>
+              val inactiveFor = millis - lastNonCometAccessedTime
+              LiftRules.sessionInactivityTimeout.vend.foreach(timeout => {
+                ApplicationLogger.warn(s"TIMEOUT: $id: $inactiveFor: $timeout ($info)")
+                if (inactiveFor > timeout) {
+                  // Ask for session termination
+                  // Race condition on destroySession that prevents destroying the underlying
+                  // HTTPSession?
+                  //info.session.destroySession()
+                  info.session.httpSession.foreach(s => s.terminate)
```

Comment:
I think we need to call `SecurityContextHolder.clearContext()` so that spring security session context is destroyed, too.